code/weapon/weapons.cpp

```diff
@@ -811,11 +811,9 @@ void parse_shockwave_info(shockwave_create_info *sci, const char *pre_char)
 
 	sprintf(buf, "%sShockwave Radius Multiplier over Lifetime Curve:", pre_char);
 	if (optional_string(buf.c_str())) {
-		SCP_string curve_name;
-		stuff_string(curve_name, F_NAME);
-		sci->radius_curve_idx = curve_get_by_name(curve_name);
+		sci->radius_curve_idx = curve_parse();
 		if (sci->radius_curve_idx < 0)
-			Warning(LOCATION, "Unrecognized shockwave radius curve '%s'", curve_name.c_str());
+			Warning(LOCATION, "Unrecognized shockwave radius curve");
```

Comment:
This `Warning` (and any `Warning` displayed as part of parsing) should be `error_display` so that the file and line number being parsed are displayed